Sources/VaporJWT/AuthError.swift

```diff
@@ -0,0 +1,89 @@
+/// Errors thrown during the parsing, creation
+/// and verification of JWT tokens by this package.
+public struct AuthError: Swift.Error {
+    public let code: Int
+    public let reasonPhrase: String
+    public let origin: Swift.Error?
+
+    init(code: Int, reasonPhrase: String, origin: Swift.Error?) {
+        self.code = code
+        self.reasonPhrase = reasonPhrase
+        self.origin = origin
+    }
+
+    public static func noAuthorizationHeader(_ origin: Swift.Error? = nil) -> Error {
```

Comment:
Oops, these methods returning `Error` was a typo. I changed to `AuthError` and the `==` works well. I can also add a test for it when I get around to those. 

`switch` works as well, though a little less concise since you have to repeat `AuthError`.

```swift
let error = AuthError.invalidBearerAuthorization()

switch error {
case AuthError.noAuthorizationHeader():
    print("foo")
case AuthError.invalidJWT():
    print("bar")
default: break
}
```

But it does enforce the default case!